spark-k8s/versions.py

```diff
@@ -1,38 +1,22 @@
 versions = [
     {
-        "product": "3.4.2",
-        "java-base": "11",
-        "java-devel": "11",
-        "python": "3.11",
-        "hadoop_long_version": "3.3.4",  # https://github.com/apache/spark/blob/1db2f5c36b120c213432fc658c9fd24fc73cb45e/pom.xml#L122
-        "aws_java_sdk_bundle": "1.12.262",  # https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws/3.3.4
-        "azure_storage": "7.0.1",  # https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-azure/3.3.4
-        "azure_keyvault_core": "1.0.0",  # https://mvnrepository.com/artifact/com.microsoft.azure/azure-storage/7.0.1
-        "jackson_dataformat_xml": "2.14.2",  # https://mvnrepository.com/artifact/org.apache.spark/spark-core_2.13/3.4.2
-        "stax2_api": "4.2.1",  # https://mvnrepository.com/artifact/com.fasterxml.jackson.dataformat/jackson-dataformat-xml/2.14.2
-        "woodstox_core": "6.5.0",  # https://mvnrepository.com/artifact/com.fasterxml.jackson.dataformat/jackson-dataformat-xml/2.14.2
-        "vector": "0.40.1",
-        "jmx_exporter": "1.0.1",
-        "tini": "0.19.0",
-    },
-    {
-        "product": "3.4.3",
-        "java-base": "11",
-        "java-devel": "11",
+        "product": "3.5.1",
+        "java-base": "17",
+        "java-devel": "17",
         "python": "3.11",
-        "hadoop_long_version": "3.3.4",  # https://github.com/apache/spark/blob/1db2f5c36b120c213432fc658c9fd24fc73cb45e/pom.xml#L122
+        "hadoop_long_version": "3.3.4",  # https://github.com/apache/spark/blob/6a5747d66e53ed0d934cdd9ca5c9bd9fde6868e6/pom.xml#L125
```

Comment:
I can't think of a hard rule that we can enforce, especially since we're updating these versions by hand. The only common sense rule I can think of is:

> ensure that any transitive dependencies of different components are compatible.

For example, Spark has a runtime dependency on aws-sdk v1. So it's probably a mistake to install aws-sdk v2 in the image.

But even this one is practically impossible to do manually without a cue on to what dependencies to check.

Reply:
> ensure that any transitive dependencies of different components are compatible.

This line is probably enough of a hint as to what to do. I'll add it in.

Thanks